test/meta-storage.test.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+const MetaStorage = require('../lib/meta-storage');
+const SinkMem = require('asset-pipe-sink-mem');
+
+test('constructor() - should take a sink', async () => {
+    const sink = new SinkMem();
+    // eslint-disable-next-line no-new
+    new MetaStorage(sink);
+});
+
+test('constructor() - should require a sink', async () => {
+    expect(() => {
+        // eslint-disable-next-line no-new
+        new MetaStorage();
+    }).toThrowError('MetaStorage needs a sink');
+});
+
+test('get() - should return null if missing entry', async () => {
+    const sink = new SinkMem();
+    const storage = new MetaStorage(sink);
+
+    const result = await storage.get('key');
+    expect(result).toBe(null);
+});
+
+test('get() - should error if non-valid json', async () => {
+    const sink = new SinkMem();
+    const storage = new MetaStorage(sink);
+
+    sink.db[storage.getKey('key')] = 'non-valid-json';
+
+    expect(storage.get('key')).rejects.toEqual(
+        new SyntaxError('Unexpected token o in JSON at position 1')
+    );
```

Comment:
Maybe it's worth handling invalid input in order to generate a better error message?

Reply:
Agree